src/lib/device/device.h

```diff
@@ -15,204 +14,167 @@
  *********************************************************************************/
 #pragma once
 
-#define BOOST_UUID_RANDOM_PROVIDER_FORCE_POSIX 1
-
-#include <algorithm>
-#include <atomic>
-#include <cstddef>
-#include <cstdint>
-#include <exception>
-#include <functional>
-#include <memory>
-#include <mutex>
-#include <string>
+#include <map>
 #include <vector>
 
-#include <boost/uuid/uuid.hpp>            // uuid class
-#include <boost/uuid/uuid_generators.hpp> // generators
-#include <boost/uuid/uuid_io.hpp>         // streaming operators etc.
+#include <isa-l/crc.h>
 #include <iomgr/iomgr.hpp>
-#include <sisl/logging/logging.h>
 #include <sisl/fds/sparse_vector.hpp>
-#include <sisl/fds/utils.hpp>
-
 #include <homestore/homestore_decl.hpp>
-
-using namespace iomgr;
-SISL_LOGGING_DECL(device, DEVICE_MANAGER)
+#include "device/hs_super_blk.h"
 
 namespace homestore {
-// forward declarations
-class BlkAllocator;
-struct blkalloc_cp;
-struct dm_info;
-struct pdevs_block;
-struct chunks_block;
-struct vdevs_block;
-struct pdev_info_block;
-struct chunk_info_block;
-struct vdev_info_block;
+
+VENUM(vdev_multi_pdev_opts_t, uint8_t, // Indicates the style of vdev when multiple pdevs are available
+      ALL_PDEV_STRIPED = 0,            // vdev data is striped across all pdevs
+      ALL_PDEV_MIRRORED = 1,           // vdev data is mirrored on all pdevs
+      SINGLE_FIRST_PDEV = 2,           // vdev data is placed only on the first pdev
+      SINGLE_ANY_PDEV = 3,             // vdev data is placed on only 1 pdev, but any of the pdev
+);
+
+#pragma pack(1)
+struct vdev_info {
+    static constexpr size_t size = 512;
+    static constexpr size_t user_private_size = 128;
+
+    uint64_t vdev_size{0};                     // 0: Size of the vdev
+    uint32_t vdev_id{0};                       // 8: Id for this vdev. It is unique per homestore instance
+    uint32_t num_mirrors{0};                   // 12: Total number of mirrors
+    uint32_t blk_size{0};                      // 16: IO block size for this vdev
+    uint32_t num_primary_chunks{0};            // 20: number of primary chunks
+    uint8_t slot_allocated{0};                 // 24: Is this current slot allocated
+    uint8_t failed{0};                         // 25: set to true if disk is replaced
+    uint8_t hs_dev_type{0};                    // 26: PDev dev type (as in fast or data)
+    uint8_t multi_pdev_choice{0};              // 27: Choice when multiple pdevs are present (vdev_multi_pdev_opts_t)
+    char name[64];                             // 28: Name of the vdev
+    uint16_t checksum{0};                      // 94: Checksum of this entire block
```

Comment:
uint16_t checksum{0}; // 92: Checksum of this entire block
so I think below will be uint8_t padding[34]{}, please double check.

Reply:
Yes adjusted thanks.